libs/events/events_test.go

```diff
@@ -166,29 +166,29 @@ func TestAddDifferentListenerForDifferentEvents(t *testing.T) {
 	}
 }
 
-var stopInputEvent = false
-var roundCount = 2000
-
-func inputRemoveListener(t *testing.T, evsw EventSwitch, done chan uint64) {
+func inputRemoveListener(t *testing.T, evsw EventSwitch, done chan uint64, roundCount int) {
 	for i := 0; i < roundCount; i++ {
 		evsw.RemoveListener("listener")
 	}
 	done <- 1
 }
 
-func inputAddListenerForEvent(t *testing.T, evsw EventSwitch, done chan uint64) {
+func inputAddListenerForEvent(t *testing.T, evsw EventSwitch, done chan uint64, roundCount int, stopInputEvent *bool) {
```

Comment:
can we use a channel instead of `*bool` for stopInputEvent

```go
func inputAddListenerForEvent(t *testing.T, evsw EventSwitch, done chan uint64, roundCount int, stopInputEvent chan<- bool) {
....
stopInputEvent <- true
```

Reply:
chan is ok， but bool flag looks more simple，it's only used for reduce fail log.